controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -601,15 +601,33 @@ func getTargetPort(service *corev1.Service, port Port) namedPort {
 	return targetPort
 }
 
+func addressChanged(oldAddress Address, newAddress Address) bool {
+	if oldAddress.Pod != nil && newAddress.Pod != nil {
+		// if these addresses are owned by pods we can check the resource versions
+		return oldAddress.Pod.ResourceVersion != newAddress.Pod.ResourceVersion
+	}
+	// in this case the identity could have changed; this can happen when for
+	// example a mirrored service is reassigned to a new gateway with a different
+	// identity and the service mirroring controller picks that and updates the
+	// identity
+	return oldAddress.Identity != newAddress.Identity
```

Comment:
Should we move this return up to before the previous one?  Currently, identity overrides only exist for mirror services, but theoretically we could have identity overrides for pod-backed services too.  In this case, we would want to return true if the identity has changed, even if the pod has not.